app/javascript/angular/code/services/_mobile_sessions.js

```diff
@@ -107,31 +107,31 @@ export const mobileSessions = (
 
         const drawSessionStartingMarker = (session, sensorName) =>
           this.drawSessionWithLabel(session, sensorName);
-        const draw = () =>
+        const draw = session =>
           drawSession.drawMobileSession(session, drawSessionStartingMarker);
         map.fitBoundsWithBottomPadding(
           calculateBounds(sensors, session, map.getZoom())
         );
-        sessionsUtils.onSingleSessionFetch(session, data, draw);
+        sessionsUtils.onSingleSessionFetch(session, draw);
       };
       params.update({ selectedSessionIds: [id] });
       this._selectSession(id, callback);
     },
 
     // this is called when refreshing a page with selected session
     reSelectSession: function(id) {
-      // this must happen before everything else, otherwise the session is drawn on the map and removed right away
-      drawSession.clear(this.sessions);
-      setTimeout(() => {
-        const callback = session => data => {
-          const drawSessionStartingMarker = (session, sensorName) =>
-            this.drawSessionWithLabel(session, sensorName);
-          const draw = () =>
-            drawSession.drawMobileSession(session, drawSessionStartingMarker);
-          sessionsUtils.onSingleSessionFetch(session, data, draw);
-        };
-        this._selectSession(id, callback);
-      }, TIMEOUT_DELAY);
```

Comment:
I removed `drawSession.clear(this.sessions);` on line 124, so I think we don't need the timeout anymore. But if you remember how it worked and how to reproduce this bug with disappearing session, let me know and I'll double check.

Reply:
Cool. I thought there was some race condition and it only occurred on slower servers or something like that.